### DISCUSSION TITLE
Km271 keine Werte

### DISCUSSION THREAD

**Discussion by Ente246** [dewenni/ESP_Buderus_KM271#86] [Q&A]
Moin, ich habe heute auch mein Board von 78mole bekommen, ich habe mich direkt für diese Firmware entschieden. Board steckt und ich komme auch auf die Web Oberfläche. Bei mir werden aber keine werte angezeigt alles steht auf 0. Kennt jemand dieses problem?
Desweiteren muss ich, wenn ich die Stromversorgung unterbreche, 5 sekunden den reset knopf drücken. Vorher passiert nichts auch keine led ist am leuchten.
Besten Dank vorab

**Comment by decanyon** [comment]
Hallo! Habe auch heute mein Board erhalten. Die erste Inbetriebnahme zeigt das identische Verhalten wie @Ente246 beschrieben hat. Bin bisher noch nicht weiter gekommen. Mit einer älteren Version der dewenni Firmware funktioniert das Board ebenfalls nicht. Morgen gehe ich in die genauere Fehleranalyse. Gruß

**Reply by Ente246** [reply]
Moin @decanyon  hast du auch das Problem, das wenn du die Heizung neu Einschaltest das, dass Board nicht Startet? Ich muss 5 Sekunden den resett Knopf drücken.
Grüße

**Reply by decanyon** [reply]
Hast du beim Aus- / Einschalten ca. 20 Sekunden gewartet? 
Ich hatte das Verhalten beobachtet, wenn der ESP beim Ausschalten nicht spannungslos ist und dann wieder Spannung bekommt. Dann startet er nicht richtig.

**Reply by Ente246** [reply]
Ich habe immer ca 3-5min gewartet bis ich wieder eingeschaltet habe. War auch zuerst meine Vermutung

**Reply by decanyon** [reply]
Kann ich gleich mal nachtesten. Auf meinem Modul ist die WebUI scheinbar abgestürzt. Werde es einmal neu starten und dabei beobachten ob es von alleine wieder hoch kommt.

**Reply by Ente246** [reply]
Ich habe es bei mir jetzt mit ca. 10min Wartezeit probiert, dann gibt es keine Probleme.

**Comment by dewenni** [comment]
Hi,

bitte mal folgendes prüfen:

**1) GPIO Einstellungen**
Hast du in den Einstellungen unter GPIO die passenden GPIO´s vorgegeben?
Hier habe ich leider in der aktuellen Version noch nicht implementiert, dass die passenden GPIO´s abhängig der Auswahl im Dropdown vorgelegt werden. Das muss man aktuell händisch eintragen.

Folgendes sollte da passen:
```
km271_RX = 4
km271_TX = 2
led_wifi = 21
heartbeat = 22
led_logmode = 17
```

**2) Log-Mode Aktiv?**
Dann kannst du bei einer funktionierenden MQTT Verbindung mal prüfen ob der "Log-Mode" aktiv ist.
Das ist der Modus in dem die Heizung die Daten an den ESP sendet. Der sollte bei korrekten Einstellungen automatisch aktiv werden.
<img width="1006" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/59ecb553-4357-4908-a85f-a96ff9110e43">

**3) Typ der Heizung**
Dann noch die Frage welchen Typ von Heizung du hast? Ist das eine Logamatic 2107?

**Reply by decanyon** [reply]
Mir hat der Punkt 1 weiter geholfen. Hatte die falschen GPIOs aus dem Schaltplan genommen. Danke!

**Reply by Ente246** [reply]
Moin, 
1) der Tipp war Gold wert, mir wurden sofort werte angezeigt.
2) Mqtt habe ich noch nicht aktiv, werde ich mich die Tage drum kümmern.
3) Habe extra nochmal geschaut ist die Logamatic 2107.

Besten Dank für die schnelle Antwort :)

Aktuell habe ich nur noch das Problem, das nach kurzer zeit die Verbindung verloren geht (ca.5-10min). Habe schon zusätzlich einen Repeater direkt bei der Heizung positioniert.

**Comment by the78mole** [comment]
Danke @dewenni 🥰, ich hatte nicht den Hauch einer Ahnung 🤣, ich bin ja nur der Hufschmied.

**Reply by dewenni** [reply]
ich werde mit der nächsten Version auch wieder mit rein nehmen, dass man abhängig von dem Dropdown die passenden GPIO Settings laden kann. Das hatte ich dummerweise vergessen auf die Version V4 entsprechend zu migrieren. Das Dropdown ist da, aber es hat derzeit keine Funktion.

<img width="750" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/558ae92d-46fb-4e08-a409-572ecf827970">

**Reply by the78mole** [reply]
Kleine Anmerkung am Rande: Auf der aktuellen Platine steht 0.0.8. Das PCB ist gleich, aber die SMD-Bestückung für den Abgassensor ist jetzt serienmäßig mit drauf. Das dient im wesentlichen mir (im Support), um die Bestückung einfacher auseinanderzuhalten. Kannst beim letzten Punkt also schreiben "0.0.7/0.0.8".

**Reply by dewenni** [reply]
ich glaube man kann auch einfach >=0.0.6 sagen. Zwischen 0.0.5 und 0.0.6 gab es eine kleine Änderung bei den LEDs. Seit dem ist es glaube ich gleich geblieben.

**Reply by the78mole** [reply]
Ich denke, zwischen 0.0.6 und 0.0.7 gab es noch die ein oder andere Option/Anpassung in der Peripherie (1-wire master, Sensor-Header,...), aber die nutzt bisher glaube niemand bzw. sind sie bei Dir dann ohnehin konfigurierbar. Also nichts, was die Grundfunktion betrifft.

**Reply by the78mole** [reply]
PS: Wir sollten uns nächstes Mal besser abstimmen... Neue Charge im Zulauf + Major FW Release gibt nur Wurstsalat (Fehlersuche) an zwei Enden 🤣

**Comment by amx1** [comment]
Vorab erst einmal ein ganz großes Dankeschön und Lob an Daniel und dewenni.

Hatte das gleiche Problem mit dem neuen Board. GPIOs eingetragen und jetzt wird das Modul erkannt und sendet per MQTT Daten an IOBROKER. Senden und steuern per MQTT oder auch per GUI funktioniert leider noch nicht. 

Gasbrenner 38kW, Steuerung ist 2107, FW 3.18

**Reply by dewenni** [reply]
@amx1 bitte auch auf [4.0.3](https://github.com/dewenni/ESP_Buderus_KM271/releases/tag/v4.0.3) updaten und dann nochmal testen.
Zum Thema ioBroker gab es hier ein paar Tipps: **MQTT via IOBROKER setzen** #58

**Reply by amx1** [reply]
@dewenni Hallo Sven, Update habe ich eingespielt und eine ext. Stromversorgung angeschlossen. Die Kommunikation per MQTT funktioniert jetzt, soweit ich getestet habe, gut. Mit node red ist Steuerung im iobroker mit mqtt Adapter super einfach.

Z.B. mit inject node eine message: topic: G234/setvalue/hk2_betriebsart und payload 0 an mqtt out node schicken, schaltet in Betriebsart Nacht um. Mit payload 1 in Betriebsart Tag. /G234/ müsst ihr natürlich entsprechend Eurer Konfig anpassen. Beim ersten mal wird das Objekt setvalue/hk2_betriebsart automatisch angelegt.

Die Objektnamen für die Umschaltung habe ich schnell im mqtt explorer rausgesucht. Gibt es eine Gesamtliste der möglichen Werte ?
(Ich bin gerade dabei meine Heizung komplett zu erneuern /WP mit WP-Konvektoren und Fussbodenheizung/ und teste, ob es sinnvoll ist den Brenner als Backup zu behalten. Mit Nacht = aus und Tag = Pufferspeicher auf Zieltemp heizen, kann ich den Brenner schon ordentlich extern steuern.

LG Axel

**Reply by decanyon** [reply]
Die Topics kannst du dir in der Header Datei _language.h_ ansehen. Brauchst du die möglichen bzw. einstellbaren Werte, sprich die payload für die Topics? 

Läuft bei dir der NTP-Sync? Mein Modul steht seit dem Update auf 01.01.1970 01:24h.

**Reply by amx1** [reply]
Danke für den Tipp
Sync auf ESP Modul ja, Übertragung an 2107 nein, (gerade noch mal getestet /Zeit manuell verstellt / sync ändert nichts

**Reply by dewenni** [reply]
Ja, der Button in für das setzen der NTP Zeit vom ESP zur Logamatic in den Einstellungen der WebUi funktioniert nicht.
Sorry, da hatte ich auch was übersehen. Per MQTT Befehl sollte es aber funktionieren.

Werde ich auch noch kurzfristig fixen.
Haben sich sich bei dem großen Umbau ein paar Fehler eingeschlichen und das ein oder andere ist vergessen gegangen.
Ich habe zugegeben danach auch noch nicht alle einzelnen Funktionen an der Heizung getestet.

**Comment by Ente246** [comment]
@dewenni Vielen Dank für das Update, seitdem läuft alles Einwandfrei. Keine Wlan Probleme mehr.

**Comment by the78mole** [comment]
Hi Axel,

> (Ich bin gerade dabei meine Heizung komplett zu erneuern /WP mit WP-Konvektoren und Fussbodenheizung/ und teste, ob es sinnvoll ist den Brenner als Backup zu behalten. Mit Nacht = aus und Tag = Pufferspeicher auf Zieltemp heizen, kann ich den Brenner schon ordentlich extern steuern.

Ich habe zwar ESPhome am Laufen, aber ich habe mir vor einiger Zeit ebenfalls meine komplette Heizungsanlage selbst umgebaut auf Wärmepumpe und Pufferspeicher. Die Ölheizung habe ich behalten, da die Wärmepumpe etwas unterdimensioniert ist (wenn saniert wurde, passt sie dann). Ich habe ein wenig herumexperimentiert mit den Betriebsarten und bei ist jetzt ist die Buderus jetzt nur noch im Warmwasser-Betriebsmodus. Den alten Warmwasserspeicher habe ich ausßer Betrieb genommen und den Fühler an den Puffer atwa mittig angelegt. Wenn die Buderus heizen soll, schalte ich Warmwasser auf Ein und wenn nicht, schalte ich aus Aus. Ganz deaktivieren lässt sich die Heizung nicht, deswegen habe ich auf Nachtbetrieb gestellt und alles auf Minimum herunter gedreht. Das scheint soweit zu funktionieren.

Grüße,
Daniel

**Reply by amx1** [reply]
Hi Daniel,

genauso habe ich das auch vor, als Backup bzw. Zusatzheizung - falls Aussen zu kalt. Eigentlich braucche ich nur drei Betriebszustände: 1) Aus = Nacht Modus auf ABSCHALT  und zweimal Tag Modus auf Zieltemp, weil ich zwei getrennte Pufferspeicher für Heizung und WW habe. Die WP ist aber noch nicht installiert. Mein Pufferspeicher schichtet sehr stark. Ich habe testweise Sensoren montiert. Unten, Mitte, Oben und Zu- und Ablauf des Brenners.

Das ist ein aktueller Screenshot.. Ich habe eine andere Steuerung. geheizt wird, wenn die obere Temp kleiner als Vorlauftemp des Heizkreises plus 5 Grad ist. Wenn ich die mittlere Temp nehmen würde, wäre der Puffer noch halb geladen.

LG Axel

![image](https://github.com/dewenni/ESP_Buderus_KM271/assets/24817084/af74d4cc-47bd-4b58-bfc4-a96e473ed596)

**Reply by the78mole** [reply]
Mit der Schichtung musst Du aber bei WP ein wenig "vorsichtiger" sein. WP haben die "dumme Angewohnheit" riesige Volumenströme zu benötigen. Dabei ist die Temperaturspreizung mit ca. 5°C sehr klein. Bedeutet im Umkehrschluss, dass man pro kW Leistung der WP einen Volumentrom von etwa 200 l/h bekommt. Erhöht man die Spreizung, geht der COP Massiv in den Keller.
Mögliche Gegenmaßnahmen sind: 

- Einen speziellen Schitspeicher verwenden, der Schichtungsrohre verwendet
- Mit Hilfe von Ventilen die WP und die Heizkreise gezielter ein- bzw. ausschichten

Ich bin da selbst noch nicht ganz am Ziel angekommen... Da fehlen noch zwei Ventile und eine kleine Steuerung. Das mache ich aber im Sommer fertig.

Grüße,
Daniel
